src/scheduler.cpp

```diff
@@ -64,13 +69,69 @@ static void generic_wait_handler(task_wait_handle wait_task)
 	thread_event.wait();
 }
 
+#if defined(EMULATE_PTHREAD_THREAD_LOCAL)
 // Wait handler function, per-thread, defaults to generic version
-static THREAD_LOCAL wait_handler thread_wait_handler = generic_wait_handler;
+static pthread_key_t thread_wait_handler_key;
```

Comment:
Wrap the static in a function, like this:
```c++
static pthread_key_t& thread_wait_handler_key()
{
	static pthread_key_t key;
	static pthread_emulation_thread_wait_handler_key_initializer initializer;
	return key;
}
```
This will ensure that the key is initialized on first use. This is necessary because Async++ may be invoked from global constructors, which might happen before the global constructors in the library have run.